src/components/StatsPanel/StatsPanel.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+import { useStatsPanel } from './StatsPanel.hook';
+import {
+  CardContainer,
+  StatItem,
+  StatIcon,
+  StatText,
+  StatValue,
+  IconWrapper,
+} from './StatsPanel.styles';
+
+import type { StatsPanelProps } from './StatsPanel.types';
+
+export const StatsPanel: React.FC<StatsPanelProps> = ({ totalTests, passed, failed, waiting }) => {
+  const stats = useStatsPanel({ totalTests, passed, failed, waiting });
+
+  return (
+    <CardContainer>
+      {stats.map((stat) => (
+        <StatItem key={stat.id}>
+          <IconWrapper>
+            <StatIcon color={stat.color}>{React.createElement(stat.icon)}</StatIcon>
```

Comment:
Using React.createElement(stat.icon) to render the icon is less idiomatic and loses type inference and the ability to pass props (for example fontSize or aria-hidden). Also stat.icon is typed as ElementType so you can render it as JSX which is clearer.

**Sugerencia:**
```typescript
Use a local variable to render the icon component in JSX and pass props for accessibility and sizing. Example:

const Icon = stat.icon;
<StatIcon color={stat.color}><Icon aria-hidden="true" fontSize="small" /></StatIcon>

Also ensure icons have aria-hidden or an accessible label depending on whether they convey information.
```